app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/disposaldetails/how_much_did_you_own.scala.html

```diff
@@ -33,10 +34,15 @@
   caption: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.caption
 )
 
-@(form: Form[ShareOfProperty], backLink: Call)(implicit request: RequestWithSessionData[_], messages:Messages, appConfig: ViewConfig)
+@(form: Form[ShareOfProperty], backLink: Call, isATrust: Boolean)(implicit request: RequestWithSessionData[_], messages:Messages, appConfig: ViewConfig)
+
+@isAgent = @{ request.userType.contains(UserType.Agent) }
+@userKey = @{if (isAgent) ".agent" else if (isATrust) ".trust" else ""}
+@userErrorPrefix =  @{if (isAgent) "agent." else if (isATrust) "trust." else ""}
```

Comment:
this is not needed in this template - we can just pass customErrorKeys to the components in here as they both accept error objects and then look for a customErrorKey - see the way components.radio_group_conditional and components.unit_input consume customErrorKey - other components might need this (when they pass in the entire error message key) but not the ones in here